packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/FIAObjectTranslator.m

```diff
@@ -470,7 +470,7 @@ + (nullable SKProductMessage *)convertProductToPigeon:(nullable SKProduct *)prod
 
   NSArray<SKProductDiscount *> *skProductDiscounts = product.discounts;
   NSMutableArray<SKProductDiscountMessage *> *pigeonProductDiscounts =
-      [[NSMutableArray alloc] init];
+      [[NSMutableArray arrayWithCapacity:skProductDiscounts.count] init];
```

Comment:
You don't want to call `init` when using `arrayWithCapacity:`, that's already taken care for you.  You just need that when you are using `alloc`.

Reply:
oh oopsies